src/neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -0,0 +1,43 @@
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using System;
+using System.Collections.Generic;
+
+namespace Neo.SmartContract.Native
+{
+    public sealed class CryptoLib : NativeContract
+    {
+        private static readonly Dictionary<NamedCurve, ECCurve> curves = new Dictionary<NamedCurve, ECCurve>
+        {
+            [NamedCurve.secp256k1] = ECCurve.Secp256k1,
+            [NamedCurve.secp256r1] = ECCurve.Secp256r1
+        };
+
+        internal CryptoLib() { }
+
+        [ContractMethod(1 << 15, CallFlags.None, Name = "ripemd160")]
+        public static byte[] RIPEMD160(byte[] data)
+        {
+            return data.RIPEMD160();
+        }
+
+        [ContractMethod(1 << 15, CallFlags.None)]
+        public static byte[] Sha256(byte[] data)
+        {
+            return data.Sha256();
+        }
+
+        [ContractMethod(1 << 15, CallFlags.None)]
+        public static bool VerifyWithECDsa(byte[] message, byte[] pubkey, byte[] signature, NamedCurve curve)
+        {
+            try
+            {
+                return Crypto.VerifySignature(message, signature, pubkey, curves[curve]);
```

Comment:
Maybe it will be faster with a `switch`?

Reply:
Maybe we will add more curves in the future.